neps/nep-0452.md

```diff
@@ -0,0 +1,168 @@
+
+## Motivation
+
+Linkdrops are an extremely powerful tool that enable seamless onboarding and instant crypto experiences with the click of a link. The original [near-linkdrop](https://github.com/near/near-linkdrop) contract provides a minimal interface allowing users to embed $NEAR within an access key and create a simple Web2 style link that can then be used as a means of onboarding. This simple $NEAR linkdrop is not enough as many artists, developers, event coordinators, and applications want to drop more digital assets such as NFTs, FTs, tickets etc.
+
+As linkdrop implementations start to push the boundaries of what’s possible, new data structures, methods, and interfaces are being developed. There needs to be a standard data model and interface put into place to ensure assets can be claimed independent of the contract they came from. If not, integrating any application with linkdrops will require customized solutions, which would become cumbersome for the developer and deteriorate the user onboarding experience. The linkdrop standard addresses these issues by providing a simple and extensible standard data model and interface.
+
+The initial discussion can be found [here](https://gov.near.org/t/official-linkdrop-standard/32463/1).
+
+## Rationale and Alternatives
+
+* Why is this design the best in the space of possible designs?
+
+This design allows for flexibility and extensibility of the standard while providing a set of criteria that cover the majority of current linkdrop use cases. The design was heavily inspired by current, functional NEPs such as the Fungible Token and Non-Fungible Token standards.
+
+* What other designs have been considered and what is the rationale for not choosing them?
+
+A generic data struct that all drop types needed to inherit from. This struct contained a name and some metadata in the form of stringified JSON. This made it easily extensible for any new types down the road.
+
+The rationale for not choosing this design was both simplicity and flexibility. Having one data struct required keys to be of one type only when in reality, they can be many at once. In addition, having a generic, open-ended metadata field could lead to many interpretations and different designs.
+
+We chose to use a KeyInfo struct that can be easily extensible and can cover all use-cases by having optional vectors of different data types. The proposed standard is simple, supports drops with multiple assets, and is backwards compatible with all previous linkdrops, and can be extended very easily.
+
+* What is the impact of not doing this?
+
+The impact of not doing this is creating a fragmented ecosystem of linkdrops, increasing the friction for user onboarding. Linkdrop claim pages (e.g. wallet providers) would have to implement custom integrations for every linkdrop provider platform. Inherently this would lead to a bad user experience when new users are onboarding and interacting with linkdrops in general.
+
+## Specification
```

Comment:
Please, add the linkdrop flow explanation here and define what is out of scope (e.g. it seems that this NEP does not define linkdrop creation interface), explain the security model, and explicitly cover the mass-creation of linkdrops (e.g. it is out of scope, though I wish it is also addressed in this NEP once and for all).

P.S. Refer to [the NEP template](https://github.com/near/NEPs/blob/master/nep-0000-template.md#specification) for more details.

Reply:
@frol Keypom has a suite of features that are all optional. We've designed the contract so that almost any use-case imaginable is covered but we don't force anyone into anything. The hashed passwords are completely optional and are used in a niche use-case (i.e ticketing). We handle mass-creation of linkdrops through hyper optimized data collections and storage. This makes it extremely cheap to create linkdrops. The only problem is that you can only execute 100 actions in a single receipt and so we can only add 100 access keys to the contract in any given call. This is a protocol level hindrance and i'm not sure how you could go about fixing that. With that being said, our model is the cheapest we've seen in the ecosystem and we've created a debitting system such that full access keys are NOT required to sign transactions and add keys thus improving the UX.

Reply:
@frol good suggestion about adding explanation.

Reply:
@frol I'v added example scenarios / flow explanation and a security implications section. Also addressed why linkdrop creation interface is out-of-scope for this NEP